terrascan.sh

```diff
@@ -47,11 +51,50 @@ function common::parse_cmdline {
   done
 }
 
+function common::is_hook_run_on_whole_repo {
+  local -a -r files=("$@")
+  local sorted_string
+  local git_ls_files
+  local included_files
+  local SCRIPT_DIR
+
+  # get directory containing this script
+  SCRIPT_DIR="$(dirname "$(realpath "${BASH_SOURCE[0]}")")"
+
+  # get included and excluded files from .pre-commit-hooks.yaml file
+  included_files=$(sed -n "/^- id: $HOOK_ID$/,/^$/p" "$SCRIPT_DIR/.pre-commit-hooks.yaml" | awk '$1 == "files:" {print $2; exit}')
+  excluded_files=$(sed -n "/^- id: $HOOK_ID$/,/^$/p" "$SCRIPT_DIR/.pre-commit-hooks.yaml" | awk '$1 == "exclude:" {print $2; exit}')
```

Comment:
Just for the sake of reducing similar calls:
```suggestion
  # get included and excluded files from .pre-commit-hooks.yaml file
  hook_config_block=$(sed -n "/^- id: $HOOK_ID$/,/^$/p" "$SCRIPT_DIR/.pre-commit-hooks.yaml")
  included_files=$(awk '$1 == "files:" {print $2; exit}' <<<"$hook_config_block")
  excluded_files=$(awk '$1 == "exclude:" {print $2; exit}' <<<"$hook_config_block")
```

Reply:
Thanks, @yermulnik good catch!